app/products/calls/actions/permissions.ts

```diff
@@ -4,24 +4,45 @@
 import {Platform} from 'react-native';
 import Permissions from 'react-native-permissions';
 
+export const hasBluetoothPermission = async () => {
+    const bluetooth = Platform.select({
+        ios: Permissions.PERMISSIONS.IOS.BLUETOOTH_PERIPHERAL,
+        default: Permissions.PERMISSIONS.ANDROID.BLUETOOTH_CONNECT,
+    });
+
+    const hasBluetooth = await Permissions.check(bluetooth);
+
+    switch (hasBluetooth) {
+        case Permissions.RESULTS.DENIED:
+        case Permissions.RESULTS.UNAVAILABLE: {
+            const permissionRequest = await Permissions.request(bluetooth);
+            return permissionRequest === Permissions.RESULTS.GRANTED;
+        }
+        case Permissions.RESULTS.BLOCKED:
+            return false;
```

Comment:
perhaps also some alert so that the user is aware and can go to the app setting to enable it? same for mic

Reply:
That's a good idea -- is there an example of where we do this in mobile currently? 
I ask because I think it's a little complicated and I'm wondering how to do it. We have the initial request (the native permissions popup), and if they deny that I'm wondering: 
1. What kind of wording we would show for the next time?
2. We wouldn't want to ask for every call, would we? (That would get annoying: "leave me alone, I denied permissions already".)
3. And if not every call, then how often do we ask for permissions?

@tanmay-des can you start thinking about these questions?

Reply:
Yes there is, maybe do a quick search in the utils folder.

Reply:
I see there are cases (e.g. `attachFileFromFiles`) that are triggered by a user requesting to do that specific thing, so it makes sense to prompt again for permissions. 
But in this case we are checking for bluetooth permissions on every call. In my opinion, asking for every call, even after they've denied it, would be annoying. Maybe even badgering... If they decide later they want to use bluetooth, they can give the permissions in the settings screen.